### PR TITLE
Init sdk bundle profiler

### PR DESCRIPTION
**Pull request by jdalmas**
Related to https://github.com/open-telemetry/opentelemetry-php/issues/567

This PR aim to  initialize & scaffold OTEL Bundle Profiler data collector.

Related issue recommands to create a DebugBundle. I don't know if it make sense or if this PR (who let profiling & debug things in OtelSdkBundle) is good enough.

Full Profiler view looks like : 
image

Bottom Profiler view looks like : 
image

### CODE REVIEW

**Comment by brettmc** [comment]
Hi @jdalmas sorry it's taken so long to get to this PR. We've short on symfony expertise at the moment, but I'll review as best I can.

**Comment by brettmc** [comment]
We have `OpenTelemetry\API\Common\Instrumentation\InstrumentationInterface` and `InstrumentationTrait` which I believe is meant to be implemented by 3rd party instrumentations like this one. Can you take a look at it and see if you think it can be incorporated?

File: src/Symfony/OtelSdkBundle/DependencyInjection/OtelSdkExtension.php [hunk 1]
```diff
@@ -432,9 +434,21 @@ private function registerSpanProcessor(Definition $definition, string $processor
             $processorType,
             $exporterKey
         );
-
         $this->getContainer()->setDefinition($id, $definition);
 
+        $env  = $this->container->getParameter('kernel.environment');
+        if (self::DEV_ENVIRONNEMENT === $env) {
```
Comment (brettmc) [inline src/Symfony/OtelSdkBundle/DependencyInjection/OtelSdkExtension.php:440]:
```suggestion
        if (self::DEV_ENVIRONMENT === $env) {
```

File: tests/Unit/Symfony/OtelSdkBundle/DataCollector/OtelDataCollectorTest.php [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+            'class' => 'SimpleSpanProcessor',
+            'file' => (new \ReflectionClass(SimpleSpanProcessor::class))->getFileName(),
+        ], $dataCollector->getData()['span_processor']);
+        $this->assertInstanceOf(Data::class, $dataCollector->getData()['resource_info_attributes']);
+        $this->assertInstanceOf(Data::class, $dataCollector->getData()['span_limits']);
+    }
+
+    public function testLateCollectWithTracerProviderSharedState(): void
+    {
+        $tracerProvider = new TracerProvider(
```
Comment (brettmc) [inline tests/Unit/Symfony/OtelSdkBundle/DataCollector/OtelDataCollectorTest.php:101]: As a general observation, the tests don't use mocking which I'd normally expect a unit test to do...

Comment (jdalmas) [reply tests/Unit/Symfony/OtelSdkBundle/DataCollector/OtelDataCollectorTest.php:101]: I'm trying to use mock but unfortunatly I did'nt success. OtelDataCollector use Reflection in order to use TracerSharedState.
And I did'nt find the right way to create a mock that work with reflection (I get `ReflectionException: Property tracerSharedState does not exist` error).
Do you have some tips for doing this kind of mock ?

Comment (brettmc) [reply tests/Unit/Symfony/OtelSdkBundle/DataCollector/OtelDataCollectorTest.php:69]: Sorry, I've never come across that one before. It's not a deal-breaker though, leave it as it is :)

**Review by brettmc — APPROVED** [review]

**Pull request merged**